pyface/action/listening_action.py

```diff
@@ -123,11 +128,10 @@ def _object_changed(self, old, new):
         for kind in ("enabled", "visible"):
             method = getattr(self, "_%s_update" % kind)
             name = getattr(self, "%s_name" % kind)
-            if name:
-                if old:
-                    old.observe(method, name, remove=True)
-                if new:
-                    new.observe(method, name)
+            if old:
+                old.observe(method, trait(name, optional=True), remove=True)
+            if new:
+                new.observe(method, trait(name, optional=True))
```

Comment:
commenting on this to draw attention to a reviewer.  Previously the observer was never getting hooked up if `name` was `None`.  This caused `pyface/action/tests/test_listening_action/TestListeningAction/test_destroy` to fail because we ultimately tried to unhook it in the destroy call.
The previous commit fixed this issue as well, but this seemed like the more correct way to go about it.